src/librbd/crypto/FormatRequest.h

```diff
@@ -0,0 +1,45 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#ifndef CEPH_LIBRBD_CRYPTO_FORMAT_REQUEST_H
+#define CEPH_LIBRBD_CRYPTO_FORMAT_REQUEST_H
+
+#include "include/rbd/librbd.hpp"
+
+struct Context;
+
+namespace librbd {
+
+class ImageCtx;
+
+namespace crypto {
+
+template <typename I>
+class FormatRequest {
```

Comment:
Nit: this seems like it would just be better implemented as a factory method? This class isn't really keeping state, it's just validating parameters and creating the actual `EncryptionFormat` class instance and invoking its format method. If the API was swapped to create the appropriate `EncryptionFormat` class first (to avoid leaking API types), the API could perform the validation and then invoke the `format` method. Since the API methods are also templated, you should be able to mock-test them as well.

Reply:
This is a breadcrumb for the next feature where Format will involve loading the parent and persisting its crypto to the child MD.

Reply:
Ack :+1: